File: backend/tests/DigitalFamilyCookbook.Tests.Utilities/Mocks/MockCategory.cs
namespace DigitalFamilyCookbook.Tests.Utilities.Mocks;

public static class MockCategory
{
    public static CategoryDto GenerateCategoryDto() => new CategoryDto
    {
        CategoryId = MockDataGenerator.RandomInteger(),
        Id = MockDataGenerator.RandomId(),
        Name = MockDataGenerator.RandomString(10, false),
    };

    public static Category GenerateCategory() => Category.FromDto(GenerateCategoryDto());

    public static List<Category> GenerateCategoryList(int count = -1)
    {
        var categortyCount = count;

        if (categortyCount < 0)
        {
            categortyCount = MockDataGenerator.RandomInteger(1, 5);
        }

        var categories = new List<Category>();

        for (var i = 0; i < categortyCount; i += 1)
        {
            categories.Add(GenerateCategory());
        }

        return categories;
    }
}

File: frontend/src/test/mocks/MockCategory.ts
import DataGenerator from '@test/DataGenerator';
import { Category } from '@models/Category';

export const MockCategory = (id = 0): Category => ({
    id: DataGenerator.GenerateGuid(),
    categoryId: id === 0 ? DataGenerator.GenerateInteger(1, 250) : id,
    name: DataGenerator.GenerateString(),
});

export const MockCategoryList = (length = 10): Category[] => {
    const categories: Category[] = [];

    for (let i = 0; i < length; i += 1) {
        categories.push(MockCategory(i + 1));
    }

    return categories;
};
